### DISCUSSION TITLE
is it possible to provide an optional arg in cmd?

### DISCUSSION THREAD

**Discussion by superiums** [gokcehan/lf#1530] [General]
is it possible to provide an optional arg in cmd?
e.g. i have the following cmd, in which the arg $1 was optional:

```bash
cmd paste-to %{{
   if [[ -n "$1" ]]; then
     cp -r --backup=numbered -i -- $fx $1
     [[ -d "$1" ]] \
       && lf -remote "send $id :unselect; cd $1;" \
       || lf -remote "send $id :unselect; select $1; " 
   else
     echo "Cancelled."
   fi
}}
map pt push :paste-to<space>
```

but it report an error while arg $1 was not provided:
```
---: line 3: $1: unbounded varable
```

**Comment by joelim-work** [comment]
Shell commands (starting with `$`/`!`/`%`/`&`) are evaluated using a shell, so you should be able to use `${1:-foo}`.

**Comment by superiums** [comment]
Thanks very much.
